packages/jest-mock/src/index.ts

```diff
@@ -20,87 +20,101 @@ export type MockFunctionMetadataType =
 export type MockFunctionMetadata<
   T,
   Y extends Array<unknown>,
-  Type = MockFunctionMetadataType,
+  MetadataType = MockFunctionMetadataType,
 > = {
   ref?: number;
   members?: Record<string, MockFunctionMetadata<T, Y>>;
   mockImpl?: (...args: Y) => T;
   name?: string;
   refID?: number;
-  type?: Type;
+  type?: MetadataType;
   value?: T;
   length?: number;
 };
 
-export type MockableFunction = (...args: Array<any>) => any;
-export type MethodKeysOf<T> = {
-  [K in keyof T]: T[K] extends MockableFunction ? K : never;
+export type ConstructorLike = {new (...args: Array<any>): any};
+
+export type MethodLike = (...args: Array<any>) => any;
+
+export type ConstructorLikeKeys<T> = {
+  [K in keyof T]: T[K] extends ConstructorLike ? K : never;
 }[keyof T];
-export type PropertyKeysOf<T> = {
-  [K in keyof T]: T[K] extends MockableFunction ? never : K;
+
+export type MethodLikeKeys<T> = {
+  [K in keyof T]: T[K] extends MethodLike ? K : never;
 }[keyof T];
 
-export type ArgumentsOf<T> = T extends (...args: infer A) => any ? A : never;
+export type PropertyLikeKeys<T> = {
```

Comment:
Alternative names: `ConstructorKeys<T>`, `MethodKeys<T>`, `PropertyKeys<T>`. The first one did not sound. Including `Like` is somewhat more clear. These utils return a tuple of keys with respectively constructor-like, method-like or property-like values of object `T`.